Black Han Sans: 1.001

aaronbell opened the pull request:
GF_CJK_Q4 work – Hotfixing the font to correct an upstream bug wherein the glyph is not properly encoded in the cmap.
Resolves: #6559 

I'm initiating this fix as a hotfix to the GF version because (1) the upstream maintainer has been [https://github.com/zesstype/Black-Han-Sans/issues/1](unresponsive on this issue) for over a year now. And (2) this version of the project was updated with a Latin character set for GF purposes, which is not reflected in the upstream repro. 

It would probably be worth (at some point) to rebuild a proper source repository for this project, but in the meantime, just leave as is.

emmamarichal commented:
Hey @aaronbell! Thank you for that PR! :)

I start the review with the fontbakery report:

## **Fontbakery**

- We have several `🔥 FAILS` related to the `Name ID6`. I think we had this issue with the previous CJK fonts, and if I remember correctly, you corrected them (but I maybe wrong, please confirm!)

🔥 FAIL Name table ID 6 (PostScript name) must be consistent across platforms.
🔥 FAIL PostScript name follows OpenType specification requirements?
🔥 FAIL Are there disallowed characters in the NAME table?
🔥 FAIL Checks METADATA.pb font.postscriptname matches postscript name declared on the name table.

- FAILS related to the description, which doesn't follow the description template. To improve it, I think we can add some more sentences and the github link -> 
`<p>To contribute, see <a href="https://github.com/zesstype/Black-Han-Sans">github.com/zesstype/Black-Han-Sans</a>.</p>`

🔥 FAIL DESCRIPTION.en_us.html must have more than 200 bytes.
🔥 FAIL Does DESCRIPTION file contain a upstream Git repo URL?
⚠️ WARN DESCRIPTION.en_us.html should end in a linebreak.
⚠️ WARN On a family update, the DESCRIPTION.en_us.html file should ideally also be updated.

- There are some issues with the way Licence and Copyright are written

🔥 FAIL METADATA.pb: Check URL on copyright string is the same as in repository_url field.
🔥 FAIL Copyright notices match canonical pattern in fonts
🔥 FAIL Check license file has good copyright string.
⚠️ WARN Check copyright namerecords match license file.
⚠️ WARN License URL matches License text on name table?

- Bonus: if we could add a designer profile for

emmamarichal commented:
@aaronbell 

Related to https://github.com/google/fonts/issues/6559:

The bug was: _When using the font "Black Hansans", the letter "몇" is displayed as "몋"._
Now, when using the font "Black Hansans", the letter "몇" doesn't seem to be supported:

<img width="647" alt="Screenshot 2024-11-27 at 11 16 48" src="https://github.com/user-attachments/assets/27976fdc-9b7d-434c-98a6-3c3f09289f98">

I want to make sure it's normal!

aaronbell commented:
Hi @emmamarichal, I reviewed the font file and decided I'd go ahead and create a proper version of uniBA87. Was able to source various bits and bobs from around the font to do so :)

I've also updated the copyright string and removed the mac platform names (which were inconsistent).

emmamarichal commented:
Hi @aaronbell,
I hotfixed all the copyright and description related fails! should be good now! :)
